Assets/GothicVR/Scripts/Creator/MeshCreator.cs

```diff
@@ -52,23 +58,38 @@ public GameObject Create(WorldData world, GameObject parent)
                 var meshFilter = subMeshObj.AddComponent<MeshFilter>();
                 var meshRenderer = subMeshObj.AddComponent<MeshRenderer>();
 
-                PrepareMeshRenderer(meshRenderer, subMesh);
+                await PrepareMeshRenderer(meshRenderer, subMesh);
+
                 PrepareMeshFilter(meshFilter, subMesh);
                 PrepareMeshCollider(subMeshObj, meshFilter.mesh, subMesh.material);
 
                 subMeshObj.SetParent(meshObj);
+
+                // Update the sub-mesh creation progress
```

Comment:
Please create a LoadingManager or ProgressbarManager so that we can have a Singleton which is handling the math within some functions.

This will lower our code on this class

Reply:
Introduced LoadingManager to handle storing progress and updating the UI to reflect the changes